### PR TITLE
feat: oauth based role management

### PR DESCRIPTION
**Pull request by Cyb4Black**
# Pull Request Checklist

### Note to first-time contributors: Please open a discussion post in [Discussions](https://github.com/open-webui/open-webui/discussions) and describe your changes before submitting a pull request.

**Before submitting, make sure you've checked the following:**

- [x] **Target branch:** Please verify that the pull request targets the `dev` branch.
- [x] **Description:** Provide a concise description of the changes made in this pull request.
- [x] **Changelog:** Ensure a changelog entry following the format of [Keep a Changelog](https://keepachangelog.com/) is added at the bottom of the PR description.
- [x] **Documentation:** Have you updated relevant documentation [Open WebUI Docs](https://github.com/open-webui/docs), or other documentation sources?
- [x] **Dependencies:** Are there any new dependencies? Have you updated the dependency versions in the documentation?
-  ~~**Testing:** Have you written and run sufficient tests for validating the changes?~~
   - Did not see any oauth-specific tests to extend for this
   - Instead manual testing in our universities environment hast taken place, testing with and without attatching open-webui to our keycloak based oauth service, with and without enabling role management
- [x] **Code review:** Have you performed a self-review of your code, addressing any coding standard issues and ensuring adherence to the project's coding standards?
- [x] **Prefix:** To cleary categorize this pull request, prefix the pull request title, using one of the following:
  - **feat**: Introduces a new feature or enhancement to the codebase
 
# Changelog Entry

### Description

- As described in the Discussion #5897 I added the ability to map roles from your oauth provider to the internal roles "user" and "admin"
- While adding this code I figured that moving oauth code to some extra file/class would make sense, so i moved oauth code to utils/oauth

### Added

- role determination function, covering either legacy role or oauth role mapping if enabled

### Changed

- moved oauth related code to util class for tightening main script

---

### Additional Information

- [docs PR is here](https://github.com/open-webui/docs/pull/246)

### Screenshots or Videos

-

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -31,6 +31,7 @@ RUN npm ci
 
 COPY . .
 ENV APP_BUILD_HASH=${BUILD_HASH}
+ENV NODE_OPTIONS="--max_old_space_size=8192"
```
Comment (tjbck) [inline Dockerfile:34]: Should be removed: https://github.com/open-webui/open-webui/pull/5639

**Comment by tjbck** [comment]
format with `npm run format:backend` wanted here, the rest seems good!

File: Dockerfile (hunk 1 above)
Comment (Cyb4Black) [reply Dockerfile]: Done

**Comment by Cyb4Black** [comment]
> format with `npm run format:backend` wanted here, the rest seems good!

Actually locally only formatted files i didn't touch for the PR.
Still want one? 😅

**Comment by tjbck** [comment]
Thanks! More testing wanted after this PR gets merged with `:dev` branch!

**Pull request merged**